### ISSUE TITLE
Error and Errors property not set when using ApolloClient.query

### ISSUE THREAD

**Issue by kartikanand** [apollographql/apollo-client#7631]
```
const client = useApolloClient();
let data = null, error = null;
try {
  const obj = await client.query({
    query: GET_ORDERS,
    variables: { after: cursor },
  });
  console.log(obj);
  data = obj.data;
  error = obj.error;
} catch (err) {
  error = err;
}
```

**Intended outcome:**
According to docs, `ApolloClient.query` function returns a promise of type `ApolloQueryResult`. This object has an `error` property for ApolloError and an `errors` property for GraphQL Errors. It seems when an actual error happens, the client just throws the error instead of setting this property. On closer inspection of the returned object, there is no error or errors properties on the object.

**Actual outcome:**
The actual outcome should've been setting the errors property with the GraphQL error (e.g. in case of throttle) instead of throwing the error.

**How to reproduce the issue:**
Create an apollo client and force a query which will throttle a graphql server.

**Versions**
  System:
    OS: macOS 11.1
  Binaries:
    Node: 15.7.0 - /usr/local/bin/node
    npm: 7.4.3 - /usr/local/bin/npm
  Browsers:
    Chrome: 88.0.4324.96
    Safari: 14.0.2
  npmPackages:
    @apollo/client: ^3.3.7 => 3.3.7 
    apollo-boost: ^0.4.4 => 0.4.7

**Comment by bradbarrow** [comment]
@kartikanand what was the resolution here? I'm having the same experience. GraphQL explorer and Apollo useQuery hooks correctly populate errors but an `apolloClient.query` directly on the server doesn't

**Comment by kartikanand** [comment]
Hey @bradbarrow , if I remember correctly this issue was happening because I was using the hook {useApolloClient} from 'react-apollo'. Once I moved to the following, I was able to get the correct errors object -

import { gql, useApolloClient } from '@apollo/client';

But since you are using this on server, maybe it could be related to some other library. Just make sure you're getting the hook/client from the correct library since there seems to be to many *-apollo libraries.
